src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/Mock/MockStream.cs

```diff
@@ -129,6 +129,64 @@ internal override async ValueTask WriteAsync(ReadOnlyMemory<byte> buffer, bool e
             }
         }
 
+        internal override ValueTask WriteAsync(ReadOnlySequence<byte> buffers, CancellationToken cancellationToken = default)
+        {
+            return WriteAsync(buffers, endStream: false, cancellationToken);
+        }
+        internal override async ValueTask WriteAsync(ReadOnlySequence<byte> buffers, bool endStream, CancellationToken cancellationToken = default)
+        {
+            CheckDisposed();
+
+            if (!_canWrite)
+            {
+                throw new NotSupportedException();
+            }
+
+            if (_socket == null)
+            {
+                await ConnectAsync(cancellationToken).ConfigureAwait(false);
+            }
+
+            foreach (ReadOnlyMemory<byte> buffer in buffers)
+            {
+                await _socket.SendAsync(buffer, SocketFlags.None, cancellationToken).ConfigureAwait(false);
+            }
+
+            if (endStream)
+            {
+                _socket.Shutdown(SocketShutdown.Send);
+            }
+        }
+
+        internal override ValueTask WriteAsync(ReadOnlyMemory<ReadOnlyMemory<byte>> buffers, CancellationToken cancellationToken = default)
+        {
+            return WriteAsync(buffers, endStream: false, cancellationToken);
+        }
+        internal override async ValueTask WriteAsync(ReadOnlyMemory<ReadOnlyMemory<byte>> buffers, bool endStream, CancellationToken cancellationToken = default)
+        {
+            CheckDisposed();
+
+            if (!_canWrite)
+            {
+                throw new NotSupportedException();
+            }
+
+            if (_socket == null)
+            {
+                await ConnectAsync(cancellationToken).ConfigureAwait(false);
+            }
+
+            foreach (ReadOnlyMemory<byte> buffer in buffers.ToArray())
+            {
+                await _socket.SendAsync(buffer, SocketFlags.None, cancellationToken).ConfigureAwait(false);
```

Comment:
`SendAsync` returns an `int bytesSent`, but hopefully we can get rid of this code soon.

Reply:
Ah yeah, my b. The code was incorrect before then I believe.

Reply:
I was considering to remove it after this change.